addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go

```diff
@@ -684,8 +684,23 @@ func (h *Helper) cloneProviderRef(
 		h.Logger.Error(err, "failed to getGVR")
 		return nil, err
 	}
+
+	newProviderName := util.GeneratePackageSecretName(cluster.Name, carvelPkgRefName) // if we find the provider in the sourceNamespace we mutate the provider name to  {CLUSTER_NAME}-{package-short-name}-package
 	provider, err := h.DynamicClient.Resource(*gvr).Namespace(sourceNamespace).Get(h.Ctx, cbPkg.ValuesFrom.ProviderRef.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) && sourceNamespace != cluster.Namespace {
+		// Did not find provider in sourceNamespace so we look in cluster namespace
+		h.Logger.Info(fmt.Sprintf("%s/%s not found. looking for %s/%s instead", sourceNamespace, cbPkg.ValuesFrom.ProviderRef.Name, cluster.Namespace, cbPkg.ValuesFrom.ProviderRef.Name))
+		provider, err = h.DynamicClient.Resource(*gvr).Namespace(cluster.Namespace).Get(h.Ctx, cbPkg.ValuesFrom.ProviderRef.Name, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			h.Logger.Info(fmt.Sprintf("%s/%s not found.", cluster.Namespace, cbPkg.ValuesFrom.ProviderRef.Name))
+		}
+		if err == nil {
+			// found provider in cluster namespace thus we don't alter name listed in clusterbootstrap
+			newProviderName = cbPkg.ValuesFrom.ProviderRef.Name
+		}
```

Comment:
How errors other than ```NotFound``` for ```h.DynamicClient.Resource.Get()```? Those are not handled here

Reply:
yes. It seems that way, but the logic actually drops an err != nil && apierros.iSNotFound(err) to line 702 
if err != nil, but I'll add an "else" to make it clear.

Reply:
hmm doing so just results in duplicate code. it looks something like 
```
	provider, err := h.DynamicClient.Resource....
	if apierrors.IsNotFound(err) && sourceNamespace ... {
		log message-1 here
		provider, err = h.DynamicClient.Resource...
		if apierrors.IsNotFound(err) {
			log message-2 here
		}
		if err != nil {.  <----- this is caught by the logic out of this loop below. 
			log message-3 here
			return nil, err
		}
		// found provider in cluster namespace, which means we should not mutate provider name.
		newProviderName = cbPkg.ValuesFrom.ProviderRef.Name
	}

       //at this point we only go on if err == nil 
	if err != nil {
		log message-3 here (again)
	}

Reply:
we'll go with this just to avoid more discussion. anyways the compiler will probably optimize this to the same resulting code

Reply:
fixed.

Reply:
turns out goimports does not like the fix. running golint we get 
```
pkg/util/clusterbootstrapclone/clusterbootstrapclone.go:670: Function 'cloneProviderRef' has too many statements (51 > 50) (funlen)
func (h *Helper) cloneProviderRef(
```

Looks like we have to go for compactnes over clarity in this case.